### PR TITLE
Fix Ruling

### PR DESCRIPTION
**Pull request by margarita-nedzelska-sonarsource**

### CODE REVIEW

**Comment by margarita-nedzelska-sonarsource** [comment]
Don't merge, I'll squash 2 first commits.

File: its/ruling/src/test/resources/expected/kotlin/ktor/kotlin-S1481.json [hunk 1]
```diff
@@ -0,0 +1,186 @@
+{
```
Comment (margarita-nedzelska-sonarsource) [inline its/ruling/src/test/resources/expected/kotlin/ktor/kotlin-S1481.json:1]: This rule is a compiler diagnostic, so all FPs are due to incomplete semantic :(

File: sonar-kotlin-plugin/src/main/java/org/sonarsource/kotlin/checks/DeprecatedCodeCheck.kt [hunk 2]
```diff
@@ -44,9 +45,11 @@ private fun KtAnnotationEntry.elementToReport(): PsiElement =
     // The first parent is always a KtDeclarationModifierList and the second parent is always an annotated entity.
     // If for some reason this assumption is broken, we report on the annotation itself.
     when (val parent = parent.parent) {
+        // Deprecated Primary constructor should always have a "constructor" keywords 
+        is KtPrimaryConstructor -> parent.getConstructorKeyword()!!
         is KtSecondaryConstructor -> parent.getConstructorKeyword()
         is KtPropertyAccessor -> parent.namePlaceholder
-        // Cannot deprecate entity without a name
+        // Cannot deprecate entity without a name (except constructors or property accessors)
```
Comment (johann-beleites-sonarsource) [inline sonar-kotlin-plugin/src/main/java/org/sonarsource/kotlin/checks/DeprecatedCodeCheck.kt:52]: Can we actually verify that this is true?

**Review by johann-beleites-sonarsource — APPROVED** [review]
LGTM!

**Pull request merged**